src/classes/RD2_DataMigrationMapper.cls

```diff
@@ -55,27 +55,185 @@ public class RD2_DataMigrationMapper {
     }
 
     /**
-     * @description Converts the RD record to RD2 format by updating the new and existing fields
-     * on the record as required.
-     * @return The updated record (or null if there is an exception?)
+     * @description Converts the RD record to enhanced Recurring Donation format 
+     * by updating the new and existing fields on the record as required.
+     * @return npe03__Recurring_Donation__c Recurring Donation in enhanced format
      */
     public npe03__Recurring_Donation__c convertToEnhancedRD() {
+        if (isConverted()) {
+            return rd;
+        }
+        
+        setInstallmentFields(rd);
+        setStartDate(rd);
+        setDayOfMonth(rd);
+        setStatus(rd);
+        setOtherFields(rd);
+
         return rd;
     }
 
+    /**	
+     * @description Returns true if the RD record is already in enhanced Recurring Donation format	
+     */	
+    public Boolean isConverted() {	
+        return String.isNotBlank(rd.Status__c) 
+            && rd.InstallmentFrequency__c != null;	
+            //frequency is null for "1st and 15th" installment period even when converted
```

Comment:
There should _always_ be a value in the Frequency field for RD2 format RD's. Though we do not have any validation or logic set to do this. It's possible that an RD inserted through an integration might not set the InstallmentFrequency field. It's probably necessary to default to this field value to 1 on the object itself so that any new records will have a value in the field; which would mean that even RD1 format RD's (created after we change the default value on the field) would have a value.

For now, let's always set the field to a value in the `setInstallmentFields()` method.

Reply:
- What would Installment Frequency be when Installment Period is "1st and 15th"?
- If we set Installment Frequency to 1 as default, legacy RD records created after the schema deployment will be considered as converted but they are not.

Reply:
- Installment Frequency will generally be 1 for 1st and 15th, but in theory it could be something else.
- Very good point on that. Status__c defaults to Active, so any RD records created after we deployed the Status__c field into orgs in the last release would have a value of "Active" in them. Let's leave Installment Frequency with a null default for now, but conversion can still set it to 1 for 1st and 15th.

Given the above, how about the following for Is Converted:
- Status__c == null
- OR (Status != 'Closed' AND {no attached Schedule records})

That would catch older RD records (the first condition) and newer RD records that haven't been converted (second condition)